allennlp/data/vocabulary.py

```diff
@@ -76,16 +76,19 @@ def __missing__(self, key: str):
         dict.__setitem__(self, key, value)
         return value
 
+    def add_non_padded_namespaces(self, non_padded_namespaces: Set[str]):
+        # add non_padded_namespaces which weren't already present
+        self._non_padded_namespaces.update(non_padded_namespaces)
 
 class _TokenToIndexDefaultDict(_NamespaceDependentDefaultDict):
-    def __init__(self, non_padded_namespaces: Sequence[str], padding_token: str, oov_token: str) -> None:
+    def __init__(self, non_padded_namespaces: Set[str], padding_token: str, oov_token: str) -> None:
         super(_TokenToIndexDefaultDict, self).__init__(non_padded_namespaces,
                                                        lambda: {padding_token: 0, oov_token: 1},
                                                        lambda: {})
 
 
 class _IndexToTokenDefaultDict(_NamespaceDependentDefaultDict):
-    def __init__(self, non_padded_namespaces: Sequence[str], padding_token: str, oov_token: str) -> None:
+    def __init__(self, non_padded_namespaces: Set[str], padding_token: str, oov_token: str) -> None:
```

Comment:
Will be called internally only: `Set[str]` vs `Iterable[str]` and `set(..)` call?